daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -954,11 +940,7 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite) {
         _.nBits
       }
 
-      val useSerializedProcessor =
-        if (validatorName == "xerces") false
-        else if (defaultValidatorName == "xerces") false
-        else if (optExpectedWarnings.isDefined) false
-        else true
+      val useSerializedProcessor = optExpectedWarnings.isEmpty
```

Comment:
Do we know why expecting warnings needs to disable serialized processors?

I guess the issue is that saved DataProcessors discard diagnostics? And I guess the TDML Runner doesn't save those diagnostics?

I wonder if we should also change that as a separate PR? It might add some extra complication since we now need to store compile diagnostics separately and recombine them into the parse/unparse diagnostics. Not sure how much it would be worth it, but it would be nice to remove this exception.